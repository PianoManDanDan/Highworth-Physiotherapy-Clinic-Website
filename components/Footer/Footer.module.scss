@use '../../styles/mixins/media-query' as *;

.footer {
  margin-top: 1rem;
  
  @include media-query(small) {
    margin-top: 2rem;
  };
  @include media-query(medium) {
    margin-top: 3rem;
  };

  &__content-wrapper {
    display: flex;
    flex-direction: column-reverse;
    justify-content: center;
    margin: 0 auto 1rem;
    max-width: 80%;

    @include media-query(small) {
      flex-direction: row;
      max-width: 66%;
    };
    @include media-query(medium) {
      max-width: 50%;
    };

    & > div {
      flex-basis: 40%;
      max-width: 70%;
      align-self: center;
      margin: 0 0.5rem;

      @include media-query(small) {
        align-self: flex-start;
      };
    };
  };
};

.social-media {
  display: flex;
  flex-direction: column;
  align-items: center;

  & > p {
    display: none;

    @include media-query(small) {
      display: block;
    };
  };

  &__links {
    display: flex;
    flex-direction: row;
    margin-top: 0.5rem;

    @include media-query(small) {
      margin-top: 0;
    };
  };
};
