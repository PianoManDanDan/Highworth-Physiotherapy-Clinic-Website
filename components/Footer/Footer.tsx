import React from 'react';
import Image from 'next/image';
import styles from './Footer.module.scss';

export const Footer: React.FC<{}> = () => {
  return (
    <footer>
      <div>
        <p>Find us on Facebook</p>
        <a 
          href="https://www.facebook.com/HighworthPhysio/"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Image
            src='/assets/logos/facebook.png'
            alt='Find us on Facebook'
            height={32}
            width={32}
            layout='intrinsic'
          />
        </a>
      </div>
      <div>
        <p>
          All of our physios are registered with at least one of the following regulatory bodies:
        </p>
        <ul>
          <li>
            <a
              href="https://www.csp.org.uk/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <abbr title="Chartered Society of Physiotherapists">CSP</abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.hcpc-uk.org/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <abbr title="Health and Care Professions Council">HCPC</abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.physiofirst.org.uk/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Physio First
            </a>
          </li>
          <li>
            <a
              href="https://www.aacp.org.uk/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <abbr title="Acupuncture Association of Chartered Physiotherapists">AACP</abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.acpat.org/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <abbr title="Association of Chartered Physiotherapists in Animal Therapy">ACPAT</abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.rampregister.org/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <abbr title="Register of Animal Musculoskeletal Practitioners">RAMP</abbr>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
}

export default Footer;
