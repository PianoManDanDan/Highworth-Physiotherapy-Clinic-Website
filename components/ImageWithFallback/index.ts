export * from './ImageWithFallback';
export * from './ImageWithFallbackProps';
