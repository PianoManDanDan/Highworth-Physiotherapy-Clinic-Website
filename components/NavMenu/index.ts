export * from './NavMenu';
export * from './subcomponents';