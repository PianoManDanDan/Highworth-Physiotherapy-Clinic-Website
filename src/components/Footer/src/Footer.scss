@import '../../../globalStyles/colors';
@import '../../../globalStyles/sizes';

.footer {
	// TODO: Fix Footer to bottom of page
	// TODO: Use white Facebook logo instead of blue
	padding-top: 2rem;
	border-top: solid 20px;
	// border-image-source: linear-gradient(90deg, color(homepage, hpc-blue, 2) 0%, color(homepage, hpc-white, 1) 100%);
	border-image: linear-gradient(0deg, color(homepage, hpc-blue, 2), color(homepage, hpc-white, 2)) 100% 0;
	border-bottom: 0;
	border-left: 0;
	border-right: 0;
	background: linear-gradient(180deg, color(homepage, hpc-blue, 2), color(homepage, hpc-gray, 1));
	color: color(homepage, hpc-white, 1);

	&__social-media {
		display: flex;
		align-items: center;
		img {
			padding: 0 1rem;
			height: 3rem;
		}
	}
	&__regulatory-bodies {
		a {
			text-decoration: none;
			color: color(homepage, hpc-white, 1);
		}

		li {
			list-style-type: none;
			padding: 0.5rem;

			&:hover {
				background: color(homepage, hpc-white, 1);
				color: color(homepage, hpc-gray, 2);
				font-weight: bold;
			}
		}
	}
	&__copyright {
		background: color(homepage, hpc-gray, 1);
		color: color(homepage, hpc-white, 1);
		font-size: 1.2rem;
	}

	.app-content-container {
		padding: 1rem 0;

		.row {
			display: flex;
			justify-content: center;
			align-items: stretch;

			@include mediaQueryFor(small, max) {
				flex-direction: column;
			}
		}
	}
}