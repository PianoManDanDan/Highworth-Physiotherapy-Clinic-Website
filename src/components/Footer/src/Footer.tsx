import React from 'react';
import { FooterProps } from './types';
import './Footer.scss';
import facebookLogo from '../../../../assets/logos/f_logo.png';

export const Footer: React.FC<FooterProps> = ({
	insuranceProviders,
	affiliateLinks,
}) => {
	return (
		<div className="footer">
			<div className="app-content-container">
				<div className="row">
					<div className="footer__facebook">
						Find us on Facebook
						<a
							href="https://www.facebook.com/HighworthPhysio/"
							target="_blank"
							rel="noopener noreferrer"
						>
							<img src={facebookLogo} alt="Find us on Facebook" />
						</a>
					</div>
				</div>
				<div className="row">
					<div className="col-md-3 col-sm-12">
						<div className="footer__heading">
								Approved by all major insurance companies:
						</div>
						<div>
							<ul>
								{insuranceProviders.map((provider) => (
									<li key={provider}>{provider}</li>
								))}
							</ul>
						</div>
					</div>
					<div className="col-md-3 col-sm-12">
						<div className="footer__heading">
								Affiliated with:
						</div>
						<div>
							<ul>
								{affiliateLinks.map((affiliate) => (
									<li key={affiliate.name}>
										<a
											href={affiliate.href}
											target="_blank"
											rel="noopener noreferrer"
										>
											{affiliate.abbrTitle ? (
												<abbr title={affiliate.abbrTitle}>
													{affiliate.name}
												</abbr>
											) : (
												affiliate.name
											)}
										</a>
									</li>
								))}
							</ul>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};
