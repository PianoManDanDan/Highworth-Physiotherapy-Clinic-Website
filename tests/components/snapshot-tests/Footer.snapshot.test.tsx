import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../../components/Footer';

describe('Footer Component snapshot test', () => {
  test('it renders all of the social media links and regulatory bodies correctly', () => {
    // Arrange
    const { asFragment } = render(
      <Footer />
    );

    // Act
    const renderedFooter = asFragment();

    // Assert
    expect(renderedFooter).toMatchSnapshot();
  });
});
