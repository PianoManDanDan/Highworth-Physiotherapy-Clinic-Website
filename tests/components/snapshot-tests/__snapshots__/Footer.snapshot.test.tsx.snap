// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component snapshot test it renders all of the social media links and regulatory bodies correctly 1`] = `
<DocumentFragment>
  <footer>
    <div>
      <div>
        <p>
          Find us on Social Media
        </p>
        <div>
          <a
            href="https://www.facebook.com/HighworthPhysio/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="32"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <p>
          All of our physios are registered with at least one of the following regulatory bodies:
        </p>
        <ul>
          <li>
            <a
              href="https://www.csp.org.uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <abbr
                title="Chartered Society of Physiotherapists"
              >
                CSP
              </abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.hcpc-uk.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <abbr
                title="Health and Care Professions Council"
              >
                HCPC
              </abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.physiofirst.org.uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Physio First
            </a>
          </li>
          <li>
            <a
              href="https://www.aacp.org.uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <abbr
                title="Acupuncture Association of Chartered Physiotherapists"
              >
                AACP
              </abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.acpat.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <abbr
                title="Association of Chartered Physiotherapists in Animal Therapy"
              >
                ACPAT
              </abbr>
            </a>
          </li>
          <li>
            <a
              href="https://www.rampregister.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <abbr
                title="Register of Animal Musculoskeletal Practitioners"
              >
                RAMP
              </abbr>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
